.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,109 @@
+name: Publish Docs If Updated
+
+on:
+  pull_request:
+    types: [closed, synchronize, opened, reopened]
+    branches:
+      - master
+
+jobs:
+  check-docs-diff:
+    runs-on: macos-latest
+    outputs:
+      diff: ${{ steps.git-diff.outputs.diff }}
+    steps:
+      - name: Checkout Vector
+        uses: actions/checkout@v2
+
+      - name: Check MD Diff
+        id: git-diff
+        uses: technote-space/get-diff-action@v3
+        with:
+          PREFIX_FILTER: |
+            docs
+
+  build:
+    runs-on: macos-latest
+    needs: check-docs-diff
+    if: ${{ needs.checkMdDiff.outputs.diff }}
+    steps:
+      - name: Clone Vector
+        uses: actions/checkout@v2
+        with:
+          repository: timberio/vector
+          ref: ${{ github.ref }}
+          path: vector
+
+      - name: Clone Vector Website
+        uses: actions/checkout@v2
+        with:
+          # timberio/vector-website and gatsby-docs respectively
+          repository: "${{ secrets.VECTOR_WEBSITE_REPO }}"
+          ref: ${{ secrets.NETLIFY_VECTOR_WEBSITE_REF }}"
+          path: vector-website
+
+      - name: Setup Cue
+        run: brew install cuelang/tap/cue
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+
+      - name: Install Dependencies
+        working-directory: ./vector-website
+        run: |
+          npm config set '//registry.npmjs.org/:_authToken' "${{ secrets.NPM_TOKEN }}"
+          yarn install
+          cd ./plugins/gatsby-plugin-graphql-codegen && yarn install
```

Comment:
@jamtur01 doesn't need to happen now, but should we consider an image for this?

Reply:
How long does it take to `yarn install` etc?

Reply:
@jamtur01 About 20 seconds for me locally